### ISSUE TITLE
Expand permissions

### ISSUE THREAD

**Issue by Carl2213** [humhub/humhub#4295]
### Is your feature request related to a problem? Please describe.
The permission system is very basic.
### Describe the solution you'd like
1. Add new permissions (on a global level):

- Permission to create public posts
- Permission to post in spaces
- Permission to comment on public posts
- Permission to use each module

2. Make permissions configurable on a per-user basis (on a global and space level)

3. Expiring permissions. After setting allow/deny, the admin can optionally add a duration for the permission. After it expires, it would reset to the opposite. (on a global and space level)

**Comment by luke-** [comment]
That's right, the current permission system is rather simple, but on the other hand very performance oriented. 

I think point 1. could be implemented relatively easily at global scope. However, this also adds a lot of complexity. For example, it would not be obvious whether a Space or Global Permission forbids the user to create a Post/Comment or use a Module. Apart from that it would be thinkable to introduce some new permissions here.

The points 2&3 are interesting but probably rather something for an individual module.

**Comment by Carl2213** [comment]
I see your point about the redundancy of having two different permissions affecting the same ability. But I still think a universal switch can be useful. Especially for the usage of modules. If a person keeps spamming global tasks, for example, that would be the only way to manage that.

The impact on performance is definitely worth considering though, if it would be noticeable.

**Comment by buddh4** [comment]
Regarding 1, I think its rather useful to change the [default container level permissions](https://docs.humhub.org/docs/admin/permissions#examples)

I'am not sure what you mean regarding Nr2, can you provide an example?

3, Is rather specific and as luke says probably rather a module candidate.

**Comment by Carl2213** [comment]
The idea behind 1 is to make it so that superadmins can control permissions, overriding container permissions. This is useful for handling spam and other kinds of unwanted content. If a person keeps spamming and gets removed from a space or has their posting permissions revoked and they go to another space and do the same. Such situations require superadmin-level action. And simply disabling the person's account is an unnecessary extreme that should be left as a final resort. The same applies to using modules. If someone is abusing a module to spam, it's pointless to prevent them from posting. You would need to revoke their permission to use that specific module.

Number 2 is a continuation of 1. Currently, if you have a problematic user, a superadmin can only control that user's permissions by making a group just for that user and controling the permissions of that group. My suggestion is to add per-user permissions. Say Mark Stevenson needs to be muted. I would click on his name in the admin panel and there would be a "permissions" tab, with the grid. Those permissions would only apply to him.

**Comment by c-nv-s** [comment]
Is there a permission setting somewhere to allow/disallow users to create spaces because I can't seem to find it???
There are situations where you only want the admins to have the ability to create new spaces so this permission would be useful

**Comment by buddh4** [comment]
@c-nv-s  `Administration -> User -> Groups` then go to dropdown permissions.

**Comment by c-nv-s** [comment]
> @c-nv-s `Administration -> User -> Groups` then go to dropdown permissions.

thanks, I knew it was hiding somewhere
